api4/command.go

```diff
@@ -98,6 +99,53 @@ func updateCommand(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(rcmd.ToJson()))
 }
 
+func moveCommand(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireCommandId()
+	if c.Err != nil {
+		return
+	}
+
+	c.LogAudit("attempt")
+
+	newTeamId, err := model.TeamIdFromCommandMoveRequestJson(r.Body)
+	if err != nil {
+		c.SetInvalidParam("team_id")
+		return
+	}
+
+	newTeam, appErr := c.App.GetTeam(newTeamId)
+	if appErr != nil {
+		c.Err = appErr
+		return
+	}
+
+	cmd, appErr := c.App.GetCommand(c.Params.CommandId)
+	if appErr != nil {
+		c.Err = appErr
```

Comment:
To follow the same approach we did on https://github.com/mattermost/mattermost-server/pull/13598, here we should do:

```suggestion
 		c.SetCommandNotFoundError() 
```

Reply:
Yes, good point. `SetCommandNotFoundError` doesn't exist yet in this branch. I'll add it.

Reply:
I updated all the command APIs to use this pattern.  It's not perfect in that is hides real permission errors and I think we need to unify the not_found and permission_error messages into one identical "command not found or no permissions".  That pattern should then be used for all APIs.  However, that really should be a new ticket.

Reply:
I think that using the "Not found" instead a "no permissions" is a quite common pattern to avoid indicating the presence of an entity. Actually, [it is documented in the HTTP RFC](https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.4) (today I learned...), so it should be safe to simply return 404 and not make any reference to permissions IMO

Reply:
I agree the HTTP status code should be 404.  I am suggesting that the accompanying text indicate that the resource was not found or user doesn't have perms.  That ways the error is identical in either case, and no information is leaked, but also clues the developer in that it *might* be a perms issue.